src/btree/bt_read.c

```diff
@@ -281,22 +204,18 @@ __las_page_instantiate(WT_SESSION_IMPL *session,
 		WT_ILLEGAL_VALUE_ERR(session);
 		}
 
+	/* The page is instantiated, we no longer need the lookaside entries. */
+	WT_ERR(__wt_las_remove_block(
+	    session, cursor, read_id, ref->page_las->las_pageid));
+
```

Comment:
`__last_page_instantiate()` can still fail without instantiating the page. Is it OK to remove the LAS entries before we know the page will be successfully instantiated?

A similar question about the caller, `__page_read()`. Currently, the last thing `__page_read()` does is call `__las_page_instantiate()`, so it can't fail after the call, however, it would be easy to add new code to that function that could fail without noticing the LAS entries had already been removed. If this is an issue, maybe a comment?

Maybe move the call to `__wt_las_remove_block()` into `__page_read()`, done after the page has been successfully instantiated.

Reply:
Sure, done.